### ISSUE TITLE
Make it straightforward to use storage besides cassandra

### ISSUE THREAD

**Issue by adriancole** [openzipkin/docker-zipkin#56]
We have folks who are trying to use mysql or redis. While cassandra is a good default, people who want to use the other two supported storage backends shouldn't need to fork in order to do that. What if we allowed someone to set `export STORAGE_TYPE=cassandra` and switch on that in the corresponding (collector|query)/run.sh? the base image could include shell scripts to source, which could set the storage env variables appropriately (as opposed to copy/paste).

**Comment by spencergibb** [comment]
:+1: I like it.

**Comment by adriancole** [comment]
ps credit to @yurishkuro for the idea to switch on `STORAGE_TYPE` as opposed to implicit

**Comment by adriancole** [comment]
ps here's roughly how to make mysql work at the moment.

```yaml
collector:
  image: openzipkin/zipkin-collector:1.13.2
  environment:
    - MYSQL_HOST=PASTE_ME
    - MYSQL_USER=PASTE_ME
    - MYSQL_PASS=PASTE_ME
  entrypoint: /bin/sh
  command: -c "java -jar /zipkin/zipkin-collector.jar -f /collector-mysql.scala"
  expose:
    - 9410
  ports:
    - 9410:9410
    - 9900:9900
query:
  image: openzipkin/zipkin-query:1.13.2
  environment:
    - MYSQL_HOST=PASTE_ME
    - MYSQL_USER=PASTE_ME
    - MYSQL_PASS=PASTE_ME
  entrypoint: /bin/sh
  command: -c "SCRIBE_HOST=${COLLECTOR_PORT_9410_TCP_ADDR} SCRIBE_PORT=9410 java -jar /zipkin/zipkin-query.jar -f /query-mysql.scala"
  expose:
    - 9411
  ports:
    - 9411:9411
    - 9901:9901
  links:
    - collector
web:
  image: openzipkin/zipkin-web:1.13.2
  ports:
    - 8080:8080
    - 9990:9990
  links:
    - collector
    - query
```

**Comment by jfeltesse-mdsol** [comment]
How would you manage the docker-compose file? I'm far from a docker specialist but if you have say `cassandra` and `mysql` containers declared it's going to pull/startup both when only one is going to actually be used.

**Comment by adriancole** [comment]
TL;DR docker-compose files are examples, and we'd expect people to make
their own for their environments. None should pull in irrelevant storage
backends.

docker-compose files are short and should be intuitive. we can have
examples for cassandra and mysql (which don't pull backends from
eachother), but I don't think we should plan to host all combinations. If
it isn't easy to throw away docker-compose files, we aren't doing a good
enough job clarifying our container config.
